### PR TITLE
correct coefficients of approximated SE3 Q_r

### PR DESCRIPTION
**Pull request by JzHuai0108**

### CODE REVIEW

**Comment by jlblancoc** [comment]
Thanks! A unit test that first fails, then passes with your fix would be great ;-)

**Comment by JzHuai0108** [comment]
Alright. I will add a unit test soon.

File: gtsam/geometry/Pose3.h [hunk 1]
```diff
@@ -181,6 +181,9 @@ class GTSAM_EXPORT Pose3: public LieGroup<Pose3, 6> {
     static Vector6 Local(const Pose3& pose, ChartJacobian Hpose = boost::none);
   };
 
+  static Matrix3 computeQforExpmapDerivative(
```
Comment (dellaert) [inline gtsam/geometry/Pose3.h:184]: static methods should be Capitalized in GTSAM

Comment (dellaert) [inline gtsam/geometry/Pose3.h:184]: Please *move* the Doxygen string from .cpp to .h. Our convention is not to duplicate: if a functions is declared in the header, the docs go there, and not in the .cpp.

File: gtsam/geometry/tests/testPose3.cpp [hunk 2]
```diff
@@ -818,6 +818,15 @@ TEST( Pose3, LogmapDerivative) {
   EXPECT(assert_equal(expectedH, actualH));
 }
 
+TEST( Pose3, computeQforExpmapDerivative) {
+  Vector6 w = Vector6::Random();
+  w.head<3>().normalize();
+  w.head<3>() = w.head<3>() * 0.09;
+  Matrix3 Qexact = Pose3::computeQforExpmapDerivative(w);
+  Matrix3 Qapprox = Pose3::computeQforExpmapDerivative(w, 0.1);
+  EXPECT(assert_equal(Qapprox, Qexact, 1e-5));
```
Comment (dellaert) [inline gtsam/geometry/tests/testPose3.cpp:827]: This function is tested against itself. How do we know the new coefficients are correct, vs the old ones?

**Review by dellaert — CHANGES_REQUESTED** [review]
Awesome. Two small nits, but a real question about the test.

File: gtsam/geometry/tests/testPose3.cpp (hunk 2 above)
Comment (JzHuai0108) [reply gtsam/geometry/tests/testPose3.cpp:827]: If the original implementation is kept but with a different function signature, then it is straightforward to show that the original implementation fails with this unit test. But I don't know if it is proper to keep the original implementation, say, in the unit test file as it looks messy. Do you have an alternative suggestion?

Comment (dellaert) [reply gtsam/geometry/tests/testPose3.cpp:827]: My suggestion would be to use numerical derivative (there are many examples on how to d- this with our specialized header).

Comment (JzHuai0108) [reply gtsam/geometry/tests/testPose3.cpp:827]: All right, Sir. As I understand, Q is one block of the Jacobian of the SE3 Expmap. As you hinted, there are two ways to compute it: the analytical one by the function we are to test, and the numeric one by the powerful numericalDerivative in gtsam. By comparing the two results, we can tell if the function is working properly. If this is what you mean, I will go ahead and write up the unit test?

Comment (dellaert) [reply gtsam/geometry/tests/testPose3.cpp:827]: Yes, a derivative check for Expmap may already exist, If so it’s just a matter of asserting that a test fails before your change and succeeds after your change. If there is no numerical derivative test yet for expmap then it should be added, and we would welcome that addition to GTSam :-)

Comment (JzHuai0108) [reply gtsam/geometry/tests/testPose3.cpp:827]: Alright. I will add it soon.

**Comment by dellaert** [comment]
Don’t you love unit tests?? :-) Is this not ready for merging in your opinion?

**Comment by JzHuai0108** [comment]
> Don’t you love unit tests?? :-) Is this now ready for merging in your opinion?

Absolutely, Sir. Unit tests are my faithful company when I am working alone on a repo. I think the pull request is ready to be merged.

**Review by dellaert — APPROVED** [review]

**Pull request merged**